packages/ui-components/src/BalanceDisplay.tsx

```diff
@@ -34,16 +33,14 @@ export function BalanceDisplay (props: BalanceDisplayProps = defaultProps) {
 
     return (
       <React.Fragment>
-        <span><b>Available:</b> <FadedText>{formatBalance(availableBalance) || PLACEHOLDER_BALANCE.toString()}</FadedText></span>
+        <span><b>Available:</b> {<FadedText>{formatBalance(availableBalance)}</FadedText> || <Loader active inline size='mini' />}</span>
         <span>
           <b>Redeemable:</b>
-          <FadedText>
-            {allStaking && allStaking.redeemable && formatBalance(allStaking.redeemable) || PLACEHOLDER_BALANCE.toString()}
-          </FadedText>
+          {allStaking && allStaking.redeemable && <FadedText>{formatBalance(allStaking.redeemable)}</FadedText>}
           {allStaking && allStaking.redeemable && allStaking.redeemable.gtn(0) && renderRedeemButton()}
         </span>
-        <span><b>Reserved:</b><FadedText>{formatBalance(reservedBalance) || PLACEHOLDER_BALANCE.toString()}</FadedText></span>
-        <span><b>Locked:</b><FadedText>{formatBalance(lockedBalance) || PLACEHOLDER_BALANCE.toString()}</FadedText></span>
+        <span><b>Reserved:</b>{ reservedBalance ? <FadedText>{formatBalance(reservedBalance)}</FadedText> : <Loader active inline size='mini' />}</span>
```

Comment:
these Loaders make for a nicer UX IMHO but console is littered with warnings about nested <p> tags...